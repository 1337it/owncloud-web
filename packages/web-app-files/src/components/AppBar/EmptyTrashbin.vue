<template>
  <action-menu-item :action="emptyTrashBinAction" :items="selectedFiles" appearance="outline" />
</template>

<script>
import { mapGetters } from 'vuex'
import ActionMenuItem from '../ActionMenuItem.vue'

import EmptyTrashBin from '../../mixins/actions/emptyTrashBin'

export default {
  name: 'EmptyTrashbin',
  components: { ActionMenuItem },
  mixins: [EmptyTrashBin],
  computed: {
    ...mapGetters('Files', ['selectedFiles']),

    emptyTrashBinAction() {
      return this.$_emptyTrashBin_items[0]
    }
  }
}
</script>
