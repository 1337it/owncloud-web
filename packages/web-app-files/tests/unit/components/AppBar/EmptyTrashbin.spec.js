import { createLocalVue, shallowMount } from '@vue/test-utils'
import EmptyTrashbin from '../../../../src/components/AppBar/EmptyTrashbin.vue'
import Files from '@/__fixtures__/files'
import VueRouter from 'vue-router'
import Vuex from 'vuex'
import DesignSystem from 'owncloud-design-system'

const localVue = createLocalVue()
localVue.use(VueRouter)
localVue.use(DesignSystem)
localVue.use(Vuex)

describe('EmptyTrashbin component', () => {
  it('should render action-menu-item if no resources are selected', () => {
    const wrapper = getMountedWrapper()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render action-menu-item for multiple selected resources', () => {
    const wrapper = getMountedWrapper([Files['/'][1], Files['/'][4]])
    expect(wrapper).toMatchSnapshot()
  })
})

function getMountedWrapper(selection = []) {
  return shallowMount(EmptyTrashbin, {
    localVue,
    store: new Vuex.Store({
      modules: {
        Files: {
          namespaced: true,
          getters: {
            selectedFiles: () => selection
          }
        }
      }
    }),
    stubs: {
      translate: true
    }
  })
}
