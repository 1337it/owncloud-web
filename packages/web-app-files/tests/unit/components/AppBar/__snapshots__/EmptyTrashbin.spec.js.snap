// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyTrashbin component should render action-menu-item for multiple selected resources 1`] = `<action-menu-item-stub action="[object Object]" items="[object Object],[object Object]" appearance="outline"></action-menu-item-stub>`;

exports[`EmptyTrashbin component should render action-menu-item if no resources are selected 1`] = `<action-menu-item-stub action="[object Object]" items="" appearance="outline"></action-menu-item-stub>`;
